### PR TITLE
fbdoc & tools update Feb 2018

### PR DESCRIPTION
**Pull request by jayrm**
fbdoc & tools update for February 2018
- wiki snapshot up to 2018-02-24
- remove dead pages
- common module for common options across (almost all) fbchkdoc tools
- additional checks in chkdocs tool to improve wiki content

### CODE REVIEW

File: doc/fbchkdoc/delextra.bas [hunk 1]
```diff
@@ -54,23 +71,67 @@ function ReadIndex( byref f as string ) as integer
...
+function ScanForDeleteMe _
+	( _
+		filename as const string _
+	) as boolean
+
+	dim sBody as string
+	dim chk as string = "!!! DELETE ME !!!"
+
+	sBody = LoadFileAsString( filename )
+	if( left( ucase( ltrim( sBody, any " " & chr(9))), len( chk ) ) = chk ) then
```
Comment (dkl) [inline doc/fbchkdoc/delextra.bas]: This `if` could be solved out, in favour of `return (left(...) = chk)`

File: doc/fbchkdoc/delextra.bas [hunk 2]
```diff
@@ -54,23 +71,67 @@ function ReadIndex( byref f as string ) as integer
...
+		loghtml( "", TRUE )
+		loghtml( "<html><body>" )
+	end if
+
 	d = dir( path + "*.wakka" )
 	while( d > "" )
+		
+		pagename = left( d, len(d) - len(".wakka") )
+
 		if( filehash.test( lcase(d) ) = FALSE ) then
```
Comment (dkl) [inline doc/fbchkdoc/delextra.bas]: maybe this `if` can be turned into a function, then e.g. `var remove = shouldRemove(d)`

File: doc/fbchkdoc/delextra.bas [hunk 3]
```diff
@@ -79,77 +140,77 @@ sub DeleteExtraFiles _
...
+end sub
 
 '' --------------------------------------------------------
 '' MAIN
 '' --------------------------------------------------------
 
-dim as string cache_dir, def_cache_dir, web_cache_dir, dev_cache_dir
-dim as integer i = 1, isgit = FALSE, nodelete = FALSE
+'' from cmd_opts.bas
+extern cmd_opt_help as boolean
```
Comment (dkl) [inline doc/fbchkdoc/delextra.bas]: `extern` declarations can be moved into a header file, like function declarations, to avoid mismatching signatures.

**Comment by jayrm** [comment]
dkl, thanks for the notes.  I appreciate you giving up your free time to do that.  I gave the delextra tool a once over and I see how I can improve the other code for next month.

**Pull request merged**